app/src/main/java/net/squanchy/service/repository/SpeakerRepository.java

```diff
@@ -27,7 +28,30 @@ public SpeakerRepository(FirebaseDbService dbService, Checksum checksum) {
                         firebaseSpeaker.id,
                         checksum.getChecksumOf(firebaseSpeaker.id),
                         firebaseSpeaker.name,
-                        firebaseSpeaker.photo_url
+                        firebaseSpeaker.bio,
+                        Optional.fromNullable(firebaseSpeaker.company_name),
+                        Optional.fromNullable(firebaseSpeaker.company_url),
+                        Optional.fromNullable(firebaseSpeaker.personal_url),
+                        Optional.fromNullable(firebaseSpeaker.photo_url),
+                        Optional.fromNullable(firebaseSpeaker.twitter_username)
                 )));
     }
+
+    public Observable<Speaker> speaker(String speakerId) {
+        return dbService.speakers()
+                .map(firebaseSpeakers -> firebaseSpeakers.speakers)
+                .flatMap(Observable::fromIterable)
+                .filter(firebaseSpeaker -> firebaseSpeaker.id.equals(speakerId))
+                .map(firebaseSpeaker -> Speaker.create(
```

Comment:
extract this and the previous one (creating a speaker from `firebaseSpeaker`) to a method